pkgs/native_assets_builder/lib/src/locking/locking.dart

```diff
@@ -56,11 +61,11 @@ Future<T> runUnderDirectoryLock<T>(
   );
 }
 
-File _fileInDir(Directory path, String filename) {
+File _fileInDir(FileSystem fileSystem, Directory path, String filename) {
```

Comment:
Use `path.fileSystem` instead, `Directory` already has a file system.

Reply:
Using `FileSystem` and `Uri` instead.

Reply:
Done.